### PR TITLE
update: search functionality and refactor

### PR DESCRIPTION
**Pull request by PuneetP16**
In this PR I have worked on the responsiveness and made search work dynamic on key press.

image

### CODE REVIEW

**Review by m26unkwn — APPROVED** [review]
Everything seems good and the code is readable.

1. Responsiveness is good.
2.  Code is clean.

File: src/components/UI/Header/Header.css [hunk 1]
```diff
@@ -102,18 +102,18 @@
...
 		display: flex;
 		justify-content: center;
-		background-color: white;
 		border-radius: 4px;
 		overflow: hidden;
-		height: 4.2rem;
+		height: 4.5rem;
 		width: 100%;
-		margin: 1rem;
+		padding: 0 1rem;
```
Comment (dev-ashishsethi) [inline src/components/UI/Header/Header.css:109]: you could have used paddiing-inline if you had to give only for left and right

**Review by dev-ashishsethi — APPROVED** [review]
good to merge

File: src/components/UI/Header/Header.css (hunk 1 above)
Comment (PuneetP16) [reply src/components/UI/Header/Header.css:109]: will look into it

**Pull request merged**